firebase-installations/src/main/java/com/google/firebase/installations/FirebaseInstallations.java

```diff
@@ -270,6 +271,19 @@ private final void doRegistration() {
         }
       }
 
+      // If persisted fid is in AUTH_TOKEN_ERROR state, clear the stored fid and update the
+      // listener with an exception.
+      if (persistedInstallationEntry.isAuthTokenErrored()) {
+        persistedInstallation.clear();
+        triggerOnException(
+            persistedInstallationEntry,
+            new FirebaseInstallationsException(
+                "Failed to generate auth token for this Firebase Installation. Call getId() "
+                    + "to recreate a new Fid and a valid auth token.",
```

Comment:
You should not only mention this important note in the exception msg.

It needs to be documented in the FirebaseInstallationsException class, basically I think there should be a specific status which indicates that the developer has to call getId() again.

Reply:
Done. PTAL.